src/org/labkey/targetedms/parser/skyaudit/AuditLogEntry.java

```diff
@@ -57,8 +63,22 @@
 
     public static AuditLogEntry retrieve(int pEntryId){
         SimpleFilter pkFilter = new SimpleFilter(FieldKey.fromParts("entryId"), pEntryId);
-        AuditLogEntry res = new TableSelector(TargetedMSManager.getTableInfoSkylineAuditLogEntry(), pkFilter, null).getObject(pEntryId, AuditLogEntry.class);
-        return res;
+        return new TableSelector(TargetedMSManager.getTableInfoSkylineAuditLogEntry(), pkFilter, null).getObject(pEntryId, AuditLogEntry.class);
+    }
+
+    public static AuditLogEntry retrieve(int pEntryId, ViewContext viewContext)
+    {
+        TargetedMSSchema schema = new TargetedMSSchema(viewContext.getUser(), viewContext.getContainer());
+        QueryDefinition qTableDef = schema.getQueryDef("AuditLogEntryContainers");
+        TableSelector sel =  new TableSelector(qTableDef.getTable(new ArrayList<>(), true));
+        Map<String, Object> params =  new HashMap<>();
+        params.put("ENTRY_ID", pEntryId);
+        params.put("CONTAINER_ID", viewContext.getContainer().getEntityId().toString());
+        sel.setNamedParameters(params);
+        if(sel.getRowCount()> 0)
```

Comment:
Could we get the full info needed to populate the AuditLogEntry back from the query, so we don't have to hit the DB twice (once to see if there's a match, and then to return the match)?